nilearn/signal.py

```diff
@@ -254,34 +254,48 @@ def _detrend(signals, inplace=False, type="linear", n_batches=10):
 
 
 def _check_wn(btype, freq, nyq):
-    wn = freq / float(nyq)
-    if wn >= 1.:
-        # results looked unstable when the critical frequencies are
-        # exactly at the Nyquist frequency. See issue at SciPy
-        # https://github.com/scipy/scipy/issues/6265. Before, SciPy 1.0.0 ("wn
-        # should be btw 0 and 1"). But, after ("0 < wn < 1"). Due to unstable
-        # results as pointed in the issue above. Hence, we forced the
-        # critical frequencies to be slightly less than 1. but not 1.
-        wn = 1 - 10 * np.finfo(1.).eps
+    """Ensure that the critical frequency works with the Nyquist frequency.
+
+    The critical frequency must be (1) >= 0 and (2) < Nyquist.
+    When critical frequencies are exactly at the Nyquist frequency,
+    results are unstable.
+
+    See issue at SciPy https://github.com/scipy/scipy/issues/6265.
+    Due to unstable results as pointed in the issue above,
+    we force the critical frequencies to be slightly less than the Nyquist
+    frequency, and slightly more than zero.
+    """
+    if freq >= nyq:
+        freq = nyq - (nyq * 10 * np.finfo(1.).eps)
         warnings.warn(
-            'The frequency specified for the %s pass filter is '
-            'too high to be handled by a digital filter (superior to '
-            'nyquist frequency). It has been lowered to %.2f (nyquist '
-            'frequency).' % (btype, wn))
+            f'The frequency specified for the {btype} pass filter is '
+            'too high to be handled by a digital filter '
+            '(superior to Nyquist frequency). '
+            f'It has been lowered to {freq} (Nyquist frequency).'
+        )
 
-    if wn < 0.0: # equal to 0.0 is okay
-        wn = np.finfo(1.).eps
+    elif freq < 0.0:  # equal to 0.0 is okay
```

Comment:
I think this condition still needs to be tested. Looks like it was never tested when initially implemented

Reply:
Good point. I'll test that.